File: python/dcc/dcc-seque.py


import cv2
import numpy as np

# Carregar imagem em escala de cinza
img = cv2.imread('imagem_4k.jpg', cv2.IMREAD_GRAYSCALE)

# Binarização (thresholding)
_, binary = cv2.threshold(img, 127, 255, cv2.THRESH_BINARY)

# Detecção de componentes conexos
num_labels, labels = cv2.connectedComponents(binary)

print(f"Número de componentes conexos encontrados: {num_labels - 1}")  # Subtrai 1 para ignorar o fundo

# (Opcional) Visualizar os componentes com cores diferentes
output = cv2.applyColorMap((labels * 255 // num_labels).astype(np.uint8), cv2.COLORMAP_JET)
cv2.imwrite("componentes_conexos_output.jpg", output)

File: c/dcc/dcc-seque.c
// componentes_conexos_seq.c
#include <opencv2/opencv.hpp>
#include <stdio.h>

int main() {
    cv::Mat img = cv::imread("imagem_4k.png", cv::IMREAD_GRAYSCALE);
    if (img.empty()) {
        printf("Erro ao carregar imagem.\n");
        return -1;
    }

    // Binariza
    cv::Mat bin;
    cv::threshold(img, bin, 127, 255, cv::THRESH_BINARY);

    // Aplica componentes conexos
    cv::Mat labels;
    int n_labels = cv::connectedComponents(bin, labels);

    printf("Número de componentes conexos: %d\n", n_labels - 1);

    // Visualização colorida
    cv::Mat colorMap;
    labels.convertTo(colorMap, CV_8U, 255.0 / n_labels);
    cv::applyColorMap(colorMap, colorMap, cv::COLORMAP_JET);
    cv::imwrite("componentes_conexos_seq_output.png", colorMap);

    return 0;
}
